src/App.tsx

```diff
@@ -67,49 +102,118 @@ export default class App extends React.Component<
+    NgElement & WithProperties<LfFieldContainerComponent>
+  >;
   fieldsService?: LfFieldsService;
-  localizationService: LfLocalizationService = new LfLocalizationService(resources);
+  localizationService: LfLocalizationService = new LfLocalizationService(
+    resources
+  );
   entrySelected: LfRepoTreeNode | undefined;
-  toolBarElement: React.RefObject<NgElement & WithProperties<LfToolbarComponent>>;
+  toolBarElement: React.RefObject<
+    NgElement & WithProperties<LfToolbarComponent>
+  >;
 
   constructor(props: any) {
     super(props);
-    this.loginComponent = React.createRef();
+    this.loginComponent = React.createRef<
+      NgElement & WithProperties<LfLoginComponent>
+    >();
     this.repositoryBrowser = React.createRef();
     this.fileInput = React.createRef();
     this.fieldContainer = React.createRef();
     this.toolBarElement = React.createRef();
-    this.setState({
+    const name: ColumnDef = {
+      id: "name",
+      displayName: "Name",
+      defaultWidth: "auto",
+      minWidthPx: 100,
+      resizable: true,
+      sortable: true,
+    };
+    const creationTime: ColumnDef = {
+      id: "creationTime",
+      displayName: "Creation Time",
+      defaultWidth: "auto",
+      resizable: true,
+      sortable: true,
+    };
+    const lastModifiedTime: ColumnDef = {
+      id: "lastModifiedTime",
+      displayName: "Last Modified Time",
+      defaultWidth: "auto",
+      resizable: true,
+      sortable: true,
+    };
+    const pageCount: ColumnDef = {
+      id: "pageCount",
+      displayName: "Page Count",
+      defaultWidth: "auto",
+      resizable: true,
+      sortable: true,
+    };
+    const templateName: ColumnDef = {
+      id: "templateName",
+      displayName: "Template Name",
+      defaultWidth: "auto",
+      resizable: true,
+      sortable: true,
+    };
+    const creator: ColumnDef = {
+      id: "creator",
+      displayName: "Author",
+      defaultWidth: "auto",
+      resizable: true,
+      sortable: true,
+    };
+    this.state = {
       expandFolderBrowser: false,
       isLoggedIn: false,
       lfSelectedFolder: undefined,
       shouldShowOpen: false,
       shouldShowSelect: false,
       shouldDisableSelect: false,
       showNewFolderDialog: false,
-      popupErrorMessage: ''
-    });
+      showEditColumnsDialog: false,
+      popupErrorMessage: "",
+      allPossibleColumns: [
+        creationTime,
+        lastModifiedTime,
+        pageCount,
+        templateName,
+        creator,
+      ],
+      selectedColumns: [name, creationTime, creator],
+    };
   }
 
   componentDidMount = async () => {
-    this.loginComponent?.current?.addEventListener('loginCompleted', this.loginCompleted);
-    this.loginComponent?.current?.addEventListener('logoutCompleted', this.logoutCompleted);
-    this.fileInput?.current?.addEventListener('change', this.selectFile);
+    this.loginComponent?.current?.addEventListener(
+      "loginCompleted",
```

Comment:
use single quotes throughout, I can send you a prettier config that should fix this on formatting the file

Reply:
I moved the prettier file to the src directory and formatted app.tsx with prettier, but it did not change the double quotes to single quotes... :/